x/ccv/provider/keeper/keeper.go

```diff
@@ -1513,3 +1514,38 @@ func (k Keeper) IsDenylistEmpty(ctx sdk.Context, chainID string) bool {
 
 	return !iterator.Valid()
 }
+
+// GetLastBondedValidators iterates the last validator powers in the staking module
+// and returns the first MaxValidators many validators with the largest powers.
+func (k Keeper) GetLastBondedValidators(ctx sdk.Context) []stakingtypes.Validator {
+	var lastPowers []stakingtypes.LastValidatorPower
+
+	maxVals := k.stakingKeeper.MaxValidators(ctx)
+
+	i := 0
+	k.stakingKeeper.IterateLastValidatorPowers(ctx, func(addr sdk.ValAddress, power int64) (stop bool) {
+		lastPowers = append(lastPowers, stakingtypes.LastValidatorPower{Address: addr.String(), Power: power})
+		i++
+		return i >= int(maxVals)
+	})
+
+	var bondedValidators []stakingtypes.Validator
+
+	for _, p := range lastPowers {
```

Comment:
Why do we need a second loop?

Reply:
I think it's preferable to have as little code as possible inside the Iterate loop, but maybe you disagree.

Concretely, the first loop just iterates over powers+addresses, and this now gets the validator objects from store

Reply:
Okay, that's fine with me! We can never be too careful with these iterators :)